aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -735,4 +765,15 @@ private Set<Tag> toTags(Map<String, String> properties, String prefix) {
   public Map<String, String> s3BucketToAccessPointMapping() {
     return s3BucketToAccessPointMapping;
   }
+
+  public <T extends S3ClientBuilder> void applyS3Configuration(T builder) {
```

Comment:
I think it feels more natural to belong to `AwsClientFactories`?

Reply:
also maybe more detailed name, like `applyS3ServiceConfigurations`?

Reply:
ah I see, with this in `AwsProperties` we can do `applyMutation(awsProperties::applyS3Configuration)` which is a clean syntax from client factory perspective, instead of `applyMutation(builder -> applyS3Configuration(awsProperties, builder))`. In that case I am fine. But I think we should rethink about other util methods in `AwsClientFactories` to see if we should move some of those as well.

Reply:
In that case, let's focus on the default client factory and assume role one for now, since they are the most commonly used, and see how it looks like if we move the util methods in `AwsClientFactories` to `AwsProperties`

Reply:
Thank you for your suggestions. Here is what I thought after investigating the code:

1. `configureEndpoint` must take different endpoints for different type of client. Hence, moving it to `AwsProperties` cannot eliminate the second param. I suggest leaving this method unchanged.
2. `credentialsProvider` must take different id, key, token for different clients. Also, the logic here is more relevant to `AwsClientFactories`. I suggest leaving this method unchanged.
3. `configureHttpClient` can be moved to `AwsProperties`. The benefit of the movement is that we can move the `httpClientType` field to `AwsProperties` so that we do not have to parse it separately in every client factory. 
However, the logic of the method seems to make it more suitable to put in the original place. 

I'll try the movement of `configureHttpClient` and see if it makes the code structure better.